service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -1,17 +1,120 @@
 package uk.gov.hmcts.reform.fpl.model;
 
-import lombok.AllArgsConstructor;
+import lombok.AccessLevel;
 import lombok.Builder;
 import lombok.Data;
+import lombok.Getter;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasDocumentStatus;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasEPOGrounds;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasMainApplicant;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasThresholdDetails;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasThresholdReason;
 
 import java.util.List;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
 @Data
-@Builder(toBuilder = true)
-@AllArgsConstructor
+@Builder
+@HasEPOGrounds
+@HasThresholdReason
+@HasThresholdDetails
+@HasDocumentStatus
+@HasMainApplicant
```

Comment:
Why is that needed? Cannot the same be achieved by checking whether collection of applicants has at least size of 1 and that applicant party is there?

Reply:
I was thinking the same yesterday, I think the error required an extra sectionType thus breaking grouping as it was scoped to Party. I'll check again.

Reply:
Confirmed, can be removed

Reply:
Please do. Generally these custom annotations should be our last resort.